python/ray/workflow/event_providers/http_event_provider.py

```diff
@@ -0,0 +1,82 @@
+import ray
+import asyncio
+import requests
+from requests import Request
+from ray import serve
+from typing import Dict, List, Optional, Any, Callable, Tuple, Union, Set
+from fastapi import FastAPI, Request
+
+
+
+class WorkflowEventHandleError(Exception):
+    def __init__(self, workflow_id: str, what_happen: str):
+        self.message = f"Workflow[id={workflow_id}] HTTP event handle failed: {what_happen}"
+        super().__init__(self.message)
+
+app = FastAPI()
+ray.init(address='auto', namespace='serve')
+serve.start(detached=True)
+
+@serve.deployment(name="myEventProvider", route_prefix="/event")
+@serve.ingress(app)
+class CustomHTTPEventProvider():
+    def __init__(self):
+        import nest_asyncio
+        nest_asyncio.apply()
+        self.event_key_payload: Dict[str, Dict[Any, Any]] = {}
+
+    # An alternative is to use
+    # @app.get("/arrivals")
+    # async def create_event_payload(self, workflow_id: str, event_key: Any, event_payload: Any):
+    @app.post("/arrivals")
+    async def create_event_payload(self, req: Request):
```

Comment:
Could you share some idea about how the fault tolerance work? For example, ACK_OK is sent, but the cluster crashed before storing it into the db.

Reply:
I feel here, we probably need to wait until checkpointed function returns before processds.

Reply:
The method `event_checkpointed` in HTTPListener is implemented with a new method `report_checkpointed` to HTTPEventProvider to confirm checkpointing. 

An alternate method `report_checkpointed_eca` will be used by event coordinator after checkpointing completed by workflow storage.

Reply:
Alternate class and the corresponding methods removed from the PR.